src/app/auth-modal/login/login.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit, OnDestroy } from "@angular/core";
+import { FormGroup, FormBuilder, Validators } from "@angular/forms";
+import { of, Subscription } from "rxjs";
+import { catchError, tap } from "rxjs/operators";
+
+import { UserService } from "src/app/core/services/user.service";
+
+import { AuthModalService } from "../auth-modal.service";
+
+@Component({
+  selector: "app-login",
+  templateUrl: "./login.component.html",
+  styleUrls: ["./login.component.less"]
+})
+export class LoginComponent implements OnInit, OnDestroy {
+  public loginForm: FormGroup;
+  public serverError: string | null = null;
```

Comment:
you don't really need to specify null as an optional type. null and undefined are always accepted as a possible value